smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -421,11 +431,16 @@ public void periodic() {
 
     // refresh the position of the robot
     this.refreshRobotPosition();
+    // update robot position with vision 
+    VisionMeasurement visionMeasurement = cameraSubsystem.getVisionPosition();
+    if (visionMeasurement.getRobotPosition() != null){
```

Comment:
While I agree we should protect the method from adding a null position ... it might be better to do that in the addVisionMeasurement method.  When that method (addVisionMeasurement) becomes private and this remains the only invocation of that method, then the null position check location becomes less of a concern.

Reply:
moved the null check to the "addVisionMeasurement" method